dts/arm/nxp/nxp_lpc55S6x.dtsi

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+		};
+		cpu@1 {
+			compatible = "arm,cortex-m33";
+			reg = <1>;
+		};
+	};
+
+	sram0:memory@20000000 {
+		compatible = "mmio-sram";
+		reg = <0x20000000 DT_SIZE_K(64)>;
+	};
+
+	sram1:memory@20010000 {
+		compatible = "mmio-sram";
+		reg = <0x20010000 DT_SIZE_K(64)>;
+	};
+
+	sram2:memory@20020000 {
+		compatible = "mmio-sram";
+		reg = <0x20020000 DT_SIZE_K(64)>;
+	};
+
+	sram3:memory@20030000 {
+		compatible = "mmio-sram";
+		reg = <0x20030000 DT_SIZE_K(64)>;
+	};
+
+	sramx:memory@40000000{
+		compatible = "mmio-sram";
+		reg = <0x40000000 DT_SIZE_K(32)>;
+	};
+
+	soc {
+
+		flash0:flash@10000000 {
+			compatible = "soc-nv-flash";
+			reg = <0x10000000 DT_SIZE_K(630)>;
+		};
+
+		usart0:usart@50086000 {
```

Comment:
we need to look at structuring this similar to how dts/arm/nordic/nrf9160* is done.  So that secure/non-secure can work cleanly in the future.

Reply:
@galak i would make a separate issue from this pull request to separate secure and non-secure configuration. I also think that it can be done different from nrf9160 because sec/nonsec HW addresses for memory and peripherals are offseted by 0x10000000.

Reply:
Ok, can we at least use the same address range for all the peripherals?